src/application_settings/container_base.py

```diff
@@ -23,13 +23,14 @@
     from typing_extensions import Self
 
 
-class ContainerBase(ContainerSectionBase, ABC):
+class ParameterContainerBase(ParameterContainerSectionBase, ABC):
```

Comment:
So why do we still need the `ParameterContainerBase` class if we moved to protocol? I feel like we now use two solutions with the same goal (defining what methods the settings / configuration have to implement).

Can we not remove all the abstract base classes in favor of the protocols?

Reply:
Yes makes sense, and clearing they both have different use cases